src/gather_classes.rs

```diff
@@ -0,0 +1,406 @@
+            // Reports may only be updated from reports with entirely different blocks.
+            return Err(anyhow!("Reports contained overlapping blocks"));
+        }
+
+        let mut updates = 0;
+        for (class_hash, table) in &report.report {
+            for (entry_point, count) in table {
+                self.update_count(*class_hash, *entry_point, *count);
+                updates += count;
+            }
+        }
+        self.blocks.extend(report.blocks.iter());
+        Ok(updates)
+    }
+
+    fn update_count(&mut self, class_hash: ClassHash, entry_point: EntryPoint, count: usize) {
+        let entry = self.report.entry(class_hash).or_default();
+        let table_entry = entry.entry(entry_point).or_insert(0);
+        *table_entry += count;
+
+        let totals_entry = self.totals.entry(class_hash).or_insert(0);
+        *totals_entry += count;
+    }
+
+    fn add_fields(&mut self, class_hash: ClassHash, entry_point: EntryPoint) {
+        self.update_count(class_hash, entry_point, 1)
+    }
+}
+
+impl core::fmt::Display for ClassHashesReport {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut map_vec = self.report.iter().collect_vec();
+        map_vec.sort_by(|(class_hash_a, _table_a), (class_hash_b, _table_b)| {
+            let count_a = self.totals[class_hash_a];
+            let count_b = self.totals[class_hash_b];
+            count_a.cmp(&count_b)
+        });
+
+        for (class_hash, table) in map_vec {
+            f.write_fmt(format_args!(
+                "class_hash: {}\tTotal: {}\n",
+                utils::felt_to_hex(class_hash, true),
+                self.totals[class_hash]
+            ))?;
+            let mut table_vec = table.iter().collect_vec();
+            table_vec.sort_by(|(entry_point_a, count_a), (entry_point_b, count_b)| {
+                match count_a.cmp(count_b) {
+                    Ordering::Less => Ordering::Less,
+                    Ordering::Greater => Ordering::Greater,
+                    Ordering::Equal => entry_point_a.cmp(entry_point_b),
+                }
+            });
+
+            f.write_fmt(format_args!("\t{: <70}\tCount\n", "Selector"))?;
+            for (entry_point, count) in table_vec {
+                f.write_fmt(format_args!(
+                    "\t{}\t{}\n",
+                    felt_to_hex(entry_point, true),
+                    count
+                ))?;
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Scans the `./results` folder for block comparisons and updates class_hashes report.
+pub async fn gather_class_hashes() -> Result<(), ManagerError> {
+    let force_generate = false;
+    let mut overall_report =
+        deserialize_or(&path_for_overall_report(), ClassHashesReport::new);
+
+    let compare_paths = get_comparison_blocks()
+        .into_iter()
+        .filter(|(block_num, _path)| !overall_report.blocks.contains(block_num))
```

Comment:
Add a method to `ClassHashReport` for looking for contained blocks and avoiding the need to interact directly with the inside field

Reply:
Added `contains_block` to `ClassHashesReport` and used that in the `filter` lambda.